### PR TITLE
Moves SchedulerJob initialization to within daemon context

### LINKED ISSUE #17120: [Scheduler error] psycopg2.OperationalError: SSL SYSCALL error: Socket operation on non-socket
Hi Airflow Team,

I am running the Airflow in an EC2 instance which is installed by conda-forge during the Code Deploy.
After upgrading the Airflow version from 2.0.2 to >=2.1.0, I am facing an error every time when I try to start the scheduler in daemon mode using this command: ```airflow scheduler --daemon```
I took a look at a similar issue #11456 and try to fix it with Python 3.8.10 and `python-daemon` 2.3.0 but still doesn't work.
The webserver is working fine, but it can't detect the scheduler.

```
Traceback (most recent call last):
  File "/home/ec2-user/anaconda3/envs/airflow_env/lib/python3.8/site-packages/sqlalchemy/engine/base.py", line 2336, in _wrap_pool_connect
    return fn()
  File "/home/ec2-user/anaconda3/envs/airflow_env/lib/python3.8/site-packages/sqlalchemy/pool/base.py", line 364, in connect
    return _ConnectionFairy._checkout(self)
  File "/home/ec2-user/anaconda3/envs/airflow_env/lib/python3.8/site-packages/sqlalchemy/pool/base.py", line 809, in _checkout
    result = pool._dialect.do_ping(fairy.connection)
  File "/home/ec2-user/anaconda3/envs/airflow_env/lib/python3.8/site-packages/sqlalchemy/engine/default.py", line 575, in do_ping
    cursor.execute(self._dialect_specific_select_one)
psycopg2.OperationalError: SSL SYSCALL error: Socket operation on non-socket
```
Relevant package version:
`sqlalchemy`=1.3.23
`psycopg2`=2.8.6
`python-daemon`=2.3.0
`apache-airflow-providers-http`=2.0.0
`apache-airflow-providers-elasticsearch`=2.0.2

### PR DESCRIPTION
**Pull request by potiuk**
In Scheduler, the SchedulerJob was instantiated before demon context was
activated. SchedulerJob is a database ORM object from SQL Alchemy and it
opens the connection to Postgres:

When you activate daemon context, what happens under the hood is forking
the process, and while some of the opened sockets were passed to the
forks (stdin and stderr but also the opened log file handle), the
established socket for DB connection was not passed.

As the result, when scheduler was started with --daemonize flag
the error `SSL SYSCALL error: Socket operation on non-socket` was
raised.

The PR moves SchedulerJob initialization to within the context
which makes the connection to Postgres initialized after the
process has been forked and daemonized.

Fixes: #17120

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/main/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/main/UPDATING.md).

### CODE REVIEW

**Comment by potiuk** [comment]
I'd love to get it merged to fix the deamonize flag :)

**Review by uranusjr — APPROVED** [review]
Analysis is sound to me

**Review by kaxil — APPROVED** [review]

File: airflow/cli/commands/scheduler_command.py [hunk 1]
```diff
@@ -29,17 +29,21 @@
 from airflow.utils.cli import process_subdir, setup_locations, setup_logging, sigint_handler, sigquit_handler
 
 
+def create_scheduler_job(args):
```
Comment (kaxil) [inline airflow/cli/commands/scheduler_command.py]:
```suggestion
def _create_scheduler_job(args):
```

nit: Probably just to suggest that it is internal function

File: airflow/cli/commands/scheduler_command.py [hunk 2]
```diff
@@ -54,9 +58,11 @@ def scheduler(args):
                 stderr=stderr_handle,
             )
             with ctx:
+                job = create_scheduler_job(args)
                 sub_proc = _serve_logs(skip_serve_logs)
                 job.run()
     else:
+        job = create_scheduler_job(args)
```
Comment (kaxil) [inline airflow/cli/commands/scheduler_command.py]:
```suggestion
        job = _create_scheduler_job(args)
```

File: airflow/cli/commands/scheduler_command.py [hunk 3]
```diff
@@ -54,9 +58,11 @@ def scheduler(args):
                 stderr=stderr_handle,
             )
             with ctx:
+                job = create_scheduler_job(args)
```
Comment (kaxil) [inline airflow/cli/commands/scheduler_command.py]:
```suggestion
                job = _create_scheduler_job(args)
```

**Pull request merged**